app/src/main/java/com/stevesoltys/seedvault/transport/backup/BackupCoordinator.kt

```diff
@@ -143,12 +143,9 @@ internal class BackupCoordinator(
         @Suppress("UNUSED_PARAMETER") isFullBackup: Boolean,
     ): Boolean {
         val packageName = targetPackage.packageName
-        // Check that the app is not blacklisted by the user
-        val enabled = settingsManager.isBackupEnabled(packageName)
-        if (!enabled) Log.w(TAG, "Backup of $packageName disabled by user.")
-        // We need to exclude the DocumentsProvider used to store backup data.
-        // Otherwise, it gets killed when we back it up, terminating our backup.
-        return enabled && targetPackage.packageName != plugin.providerPackageName
+        val shouldInclude = packageService.shouldIncludeAppInBackup(packageName)
+        if (!shouldInclude) Log.i(TAG, "Excluding $packageName from backup.")
+        return shouldInclude
```

Comment:
We should test apps with lots of data first, but going over quota means app doesn't get backed up at all. So I'd auto-disable quota when d2d is toggled on, right in the expert settings fragment. can even be a dependency in XML.

Reply:
I changed the logic - if the user enables D2D backups, unlimited quota will also be enabled automatically.

Reply:
Did you push this? Didn't see this in the push diffs github gives me.

Reply:
Strange, I found it in the overall diff. Looks like some incremental diffs are missing :(
Why not use the XML dependency method instead of manually checking the unlimited quota setting?

Reply:
When done this way it doesn't require that D2D is enabled for the user to enable unlimited quota.